python/lsst/obs/base/fitsRawFormatterBase.py

```diff
@@ -193,62 +162,64 @@ def readInfoComponent(self, fileDescriptor, component):
+            return self.makeFilter()
         elif component == "visitInfo":
-            return self.makeVisitInfo(self.readMetadata(fileDescriptor))
+            return self.makeVisitInfo()
         elif component == "wcs":
-            return self.makeWcs(self.readMetadata(fileDescriptor))
+            return self.makeWcs()
         return None
 
-    def readFull(self, fileDescriptor, parameters=None):
+    def readFull(self, parameters=None):
         """Read the full Exposure object.
 
         Parameters
         ----------
-        fileDescriptor : `FileDescriptor`
-            Identifies the file to read and parameters to be used for reading.
         parameters : `dict`, optional
             If specified a dictionary of slicing parameters that overrides
-            those in ``fileDescriptor`.
+            those in ``self.fileDescriptor`.
 
         Returns
         -------
         exposure : `~lsst.afw.image.Exposure`
             Complete in-memory exposure.
         """
         from lsst.afw.image import makeExposure, makeMaskedImage
-        full = makeExposure(makeMaskedImage(self.readImage(fileDescriptor)))
-        mask = self.readMask(fileDescriptor)
+        full = makeExposure(makeMaskedImage(self.readImage()))
+        mask = self.readMask()
         if mask is not None:
             full.setMask(mask)
-        variance = self.readVariance(fileDescriptor)
+        variance = self.readVariance()
         if variance is not None:
             full.setVariance(variance)
-        metadata = self.readMetadata(fileDescriptor)
         info = full.getInfo()
-        info.setWcs(self.makeWcs(metadata))
-        info.setFilter(self.makeFilter(metadata))
-        info.setVisitInfo(self.makeVisitInfo(metadata))
+        info.setWcs(self.makeWcs())
+        info.setFilter(self.makeFilter())
+        info.setVisitInfo(self.makeVisitInfo())
         # We shouldn't have to call stripMetadata() here because that should
         # have been done by makeVisitInfo and makeWcs (or by subclasses that
         # strip metadata for other components when constructing them).
-        full.setMetadata(metadata)
+        full.setMetadata(self.metadata)
         return full
 
-    def write(self, inMemoryDataset, fileDescriptor):
+    def write(self, inMemoryDataset):
         """Write a Python object to a file.
 
         Parameters
         ----------
         inMemoryDataset : `object`
             The Python object to store.
-        fileDescriptor : `FileDescriptor`
-            Identifies the file to read, type to read it into and parameters
-            to be used for reading.
 
         Returns
         -------
         path : `str`
             The `URI` where the primary file is stored.
         """
         raise NotImplementedError("Raw data cannot be `put`.")
+
+    @property
+    def observationInfo(self):
+        """The ObservationInfo extracted from this file's metadata.
```

Comment:
Can you link that type specifier? https://developer.lsst.io/python/numpydoc.html#documenting-class-properties